OpenRA.Mods.Common/Traits/Taggable.cs

```diff
@@ -0,0 +1,57 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Allows this actor to be 'tagged' with arbitrary strings. Tags must be unique or they will be rejected.")]
+	public class TaggableInfo : UsesInit<TagsInit>
+	{
+		object ITraitInfo.Create(ActorInitializer init) { return new Taggable(init, this); }
+	}
+
+	public class Taggable
```

Comment:
How about we name this `ScriptTags` to match e.g. `ScriptTriggers` and `ScriptUpgradesCache`?